### PR TITLE
ZIOS-11478: Implement second device alert

### PR DESCRIPTION
**Pull request by alexaubry**
### Issues

- We want to notify users on all devices when legal hold is disabled
- We want to notify users on their other devices that legal hold was accepted

### Solutions

- Implement the alert for legal hold on second device

### Bugs Fixed

We also fix a bug where the conversation list header wouldn't observe changes, and where it would show the wrong shape for team users after logging in.

### To-Do

- [x] Show the pending request alert when coming to the foreground
- [x] ACK legal hold status when the user taps the disabled alert
- [x] Fix bug where the "legal hold is disabled" alert is shown after the user accepts legal hold

### CODE REVIEW

File: Wire-iOS/Sources/Managers/LegalHoldDisclosureController.swift [hunk 1]
```diff
@@ -0,0 +1,224 @@
...
+class LegalHoldDisclosureController: NSObject, ZMUserObserver {
+
+    enum DisclosureState {
+        /// No legal hold status is being disclosed.
+        case none
+
+        /// The user is being warned about a pending legal hold alert.
+        case warningAboutPendingRequest(UIAlertController)
+
+        ///
```
Comment (typfel) [inline Wire-iOS/Sources/Managers/LegalHoldDisclosureController.swift]: Incomplete comment?

Comment (alexaubry) [reply Wire-iOS/Sources/Managers/LegalHoldDisclosureController.swift]: Added the missing comments, thanks

**Comment by typfel** [comment]
Since you deleted `UIViewController+LegalHoldAlertSnapshotTests.swift` shouldn't there also be some associated snapshots to delete?

File: Wire-iOS/Sources/AppRootViewController.swift [hunk 2]
```diff
@@ -45,6 +45,7 @@ final class AppRootViewController: UIViewController {
     fileprivate let mediaManagerLoader = MediaManagerLoader()
 
     var authenticationCoordinator: AuthenticationCoordinator?
+    var legalHoldDisclosureController: LegalHoldDisclosureController?
```
Comment (typfel) [inline Wire-iOS/Sources/AppRootViewController.swift]: Why does the `LegalHoldDisclosureController` live in the `AppRootViewController` and not in the `ZClientViewController`? Wouldn't the need for `resetLegalHoldDisclosureControllerIfNeeded()` go away if was created & destroyed together with the `ZClientViewController`?

Comment (alexaubry) [reply Wire-iOS/Sources/AppRootViewController.swift]: That's a good point, thanks.

Comment (alexaubry) [reply Wire-iOS/Sources/AppRootViewController.swift]: Moved to the client view controller. Does it look good to you now?

**Comment by billypchan** [comment]
snapshot test fails due to title bar is overflown. Maybe caused by the addition of top right indicator?

Screenshot 2019-06-20 at 10 38 21

**Comment by billypchan** [comment]
Another snapshot failure is related to mockusers name, Tarja is the last user in `people-01.json` and James is the first one. 

Screenshot 2019-06-20 at 10 49 57

**Comment by alexaubry** [comment]
@billypchan I couldn't check the snapshots before now because I was away, but thanks for the input!

**Comment by typfel** [comment]
@billypchan are you ok with this now?

**Pull request merged**